packages/gestalt-charts/src/ChartGraph.js

```diff
@@ -0,0 +1,586 @@
+
+  const referenceAreaSummary = referenceAreas
+    ? referenceAreas.map(({ label, style }) => ({
+        label,
+        style,
+      }))
+    : null;
+
+  const customTooltip = useCustomTooltip({
+    isDarkMode,
+    renderTooltip,
+  });
+
+  const defaultTooltip = useDefaultTooltip({ isDarkMode, labelMap, tickFormatter, isTimeSeries });
+
+  const defaultLegend = useDefaultLegend({
+    isHorizontalBiaxial,
+    isVerticalBiaxial,
+    height: chartHeight,
+    legend,
+    labelMap,
+    setLegendHeight,
+    referenceAreaSummary,
+  });
+
+  return (
+    <ChartProvider decal={visualPatternSelected}>
+      <Box
+        width={isHorizontal ? '100%' : undefined}
+        display="flex"
+        direction="column"
+        color="default"
+        padding={4}
+      >
+        {visualPatternSelected !== 'disabled' || title ? (
+          <Header
+            readyToRender={chartWidth > 0}
+            title={title}
+            description={description}
+            onVisualPatternChange={onVisualPatternChange}
+            visualPatternSelected={visualPatternSelected}
+          />
+        ) : null}
+        {selectors && selectors.selector === 'TileData' ? (
+          <Box marginBottom={4}>
+            <Flex gap={2}>
+              {selectors.data.map((tile) => (
+                <TileData
+                  key={tile.id}
+                  id={tile.id}
+                  color={tile.color}
+                  title={tile.title}
+                  value={tile.value}
+                  selected={tile.selected}
+                  onTap={tile.onTap}
+                  trend={tile.trend}
+                  disabled={tile.disabled}
+                  showCheckbox={tile.showCheckbox}
+                  tooltip={tile.tooltip}
+                  trendSentiment={tile.trendSentiment}
+                />
+              ))}
+            </Flex>
+          </Box>
+        ) : null}
+        {selectors && selectors.selector === 'TagData' ? (
+          <Box marginBottom={4}>
+            <Flex gap={2}>
+              {selectors.data.map((tag) => (
+                <TagData
+                  key={tag.id}
+                  accessibilityRemoveIconLabel={tag.accessibilityRemoveIconLabel}
+                  baseColor={tag.baseColor}
+                  color={tag.color}
+                  disabled={tag.disabled}
+                  id={tag.id}
+                  onTap={tag.onTap}
+                  onRemove={tag.onRemove}
+                  selected={tag.selected}
+                  showCheckbox={tag.showCheckbox}
+                  size={tag.size}
+                  text={tag.text}
+                  tooltip={tag.tooltip}
+                />
+              ))}
+            </Flex>
+          </Box>
+        ) : null}
+        <div className="_gestalt" style={{ width: '100%', height: '100%', maxWidth: 960 }}>
+          <ResponsiveContainer
+            debounce={150}
+            onResize={(width, height) => {
+              setChartHeight(height);
```

Comment:
I was looking at the `readyToRender` stuff. What if you moved this into the `ChartProvider` and then all of the children will have access to height.

It might be more re-usable as we add more components, and help us responsively adapt the other components

```
const {width, height} = useChartProvider()

if(width < 0 || height < 0) 
```